user/src/com/google/gwt/user/client/ui/RootPanel.java

```diff
@@ -223,16 +232,36 @@ public static native com.google.gwt.user.client.Element getBodyElement() /*-{
 
   /**
    * Determines whether the given widget is in the detach list.
-   * 
+   * <p>
+   * Note that modern browsers do not have the memory leaks that originally required use of this
+   * feature - it is retained only to support application-specific detach events.
+   * </p>
+   *
    * @param widget the widget to be checked
    * @return <code>true</code> if the widget is in the detach list
+   * @deprecated Instead, prefer to use the {@code bfcache}, {@code pagehide},
+   * {@code visibilitychange} events, etc.
    */
+  @Deprecated
   public static boolean isInDetachList(Widget widget) {
     return widgetsToDetach.contains(widget);
   }
 
-  // Package-protected for use by unit tests. Do not call this method directly.
-  static void detachWidgets() {
+  /**
+   * Detaches all widgets that were set to be detached on window close by a call to
+   * {@link #detachOnWindowClose}. Formerly was package-protected, now can be called by projects
+   * that required the old behavior and are willing to set up their own onclose handler on the
+   * window.
+   * <p>
+   * Note that modern browsers do not have the memory leaks that originally required use of this
+   * feature - it is retained only to support application-specific detach events.
+   * </p>
+   *
+   * @deprecated Instead, prefer to use the {@code bfcache}, {@code pagehide},
```

Comment:
`@deprecated See deprecation notice in {@link #detachOnWindowClose} for further information.`